### PR TITLE
Remove all __IPHONE_11_0 checks now that we no longer support Xcode 8.

### PR DESCRIPTION
**Pull request by jverkoey**
We can now make use of @available throughout our codebase.

We support Xcode 9 and up, which includes the iOS 11 SDK. This means we can remove any guards for SDKs prior to iOS 11.

This was cleaned up by running a global find-and-replace with the following regular expression:

```
Find:#if defined\(__IPHONE_11_0\) && \(__IPHONE_OS_VERSION_MAX_ALLOWED >= __IPHONE_11_0\)\n(.+if \(@available\(iOS 11.0, \*\)\) \{(?:.|\n)*?)(?:#else(?:.|\n)*?)?\n#endif
Replace:$1
```

With some additional cleanup for stragglers that didn't match this pattern. Note that else clauses were intentionally dropped.

Closes https://github.com/material-components/material-components-ios/issues/4909

### CODE REVIEW

**Review by ianegordon — APPROVED** [review]
Thanks you @jverkoey !

File: components/Buttons/examples/supplemental/ButtonsTypicalUseSupplemental.m [hunk 1]
```diff
@@ -91,16 +91,13 @@ - (CGRect)contentBounds {
...
   if (@available(iOS 11.0, *)) {
       UIEdgeInsets safeAreaInsets = self.view.safeAreaInsets;
       contentBounds = UIEdgeInsetsInsetRect(bounds, safeAreaInsets);
   } else {
     preiOS11Behavior();
   }
-#else
-  preiOS11Behavior();
-#endif
+
```
Comment (romoore) [inline components/Buttons/examples/supplemental/ButtonsTypicalUseSupplemental.m]: Nit: Remove extra newline?

File: components/FlexibleHeader/src/MDCFlexibleHeaderView.m [hunk 2]
```diff
@@ -388,15 +388,12 @@ - (void)extractTopSafeAreaInset {
...
-#if defined(__IPHONE_11_0) && (__IPHONE_OS_VERSION_MAX_ALLOWED >= __IPHONE_11_0)
     if (@available(iOS 11.0, *)) {
       self.topSafeAreaInset = viewController.view.safeAreaInsets.top;
     } else {
       self.topSafeAreaInset = viewController.topLayoutGuide.length;
     }
-#else
-    self.topSafeAreaInset = viewController.topLayoutGuide.length;
-#endif
+
```
Comment (romoore) [inline components/FlexibleHeader/src/MDCFlexibleHeaderView.m]: Nit: Remove extra newline

File: components/FlexibleHeader/src/MDCFlexibleHeaderView.m [hunk 3]
```diff
@@ -624,14 +617,11 @@ - (void)fhv_removeInsetsFromScrollView:(UIScrollView *)scrollView {
...
 
-#if defined(__IPHONE_11_0) && (__IPHONE_OS_VERSION_MAX_ALLOWED >= __IPHONE_11_0)
   if (@available(iOS 11.0, *)) {
     existingContentInsetAdjustment = (scrollView.adjustedContentInset.top
                                       - scrollView.contentInset.top);
   }
-#else
-  (void)scrollView; // To silence unused variable warnings.
-#endif
+
```
Comment (romoore) [inline components/FlexibleHeader/src/MDCFlexibleHeaderView.m]: Nit: Remove extra newline

File: components/TextFields/examples/MultilineTextFieldLegacyExample.m [hunk 4]
```diff
@@ -197,24 +196,7 @@ - (void)viewDidLoad {
...
-    [NSLayoutConstraint constraintWithItem:multilineTextFieldCharMaxFullWidth
-                                 attribute:NSLayoutAttributeBottom
-                                 relatedBy:NSLayoutRelationEqual
-                                    toItem:self.scrollView
-                                 attribute:NSLayoutAttributeBottomMargin
-                                multiplier:1
-                                  constant:-20]
-  ]];
-#endif
+
```
Comment (romoore) [inline components/TextFields/examples/MultilineTextFieldLegacyExample.m]: Nit: Remove extra newline

File: components/TextFields/examples/TextFieldCustomFontExample.m [hunk 5]
```diff
@@ -138,24 +137,7 @@ - (void)viewDidLoad {
...
-                                            [NSLayoutConstraint constraintWithItem:systemFontTextField
-                                                                         attribute:NSLayoutAttributeBottom
-                                                                         relatedBy:NSLayoutRelationEqual
-                                                                            toItem:self.scrollView
-                                                                         attribute:NSLayoutAttributeBottomMargin
-                                                                        multiplier:1
-                                                                          constant:-20]
-                                            ]];
-#endif
+
```
Comment (romoore) [inline components/TextFields/examples/TextFieldCustomFontExample.m]: Nit: Remove extra newline

File: components/TextFields/examples/TextFieldOutlinedExample.m [hunk 6]
```diff
@@ -246,24 +245,7 @@ - (void)viewDidLoad {
...
-    [NSLayoutConstraint constraintWithItem:textFieldMessage
-                                 attribute:NSLayoutAttributeBottom
-                                 relatedBy:NSLayoutRelationEqual
-                                    toItem:self.scrollView
-                                 attribute:NSLayoutAttributeBottomMargin
-                                multiplier:1
-                                  constant:-20]
-  ]];
-#endif
+
```
Comment (romoore) [inline components/TextFields/examples/TextFieldOutlinedExample.m]: Nit: Remove extra newline

File: components/TextFields/examples/TextFieldStatesViewController.m [hunk 7]
```diff
@@ -151,24 +150,7 @@ - (void)viewDidLoad {
...
-  [NSLayoutConstraint constraintWithItem:self.erroredDisabled
-                               attribute:NSLayoutAttributeBottom
-                               relatedBy:NSLayoutRelationEqual
-                                  toItem:self.scrollView
-                               attribute:NSLayoutAttributeBottomMargin
-                              multiplier:1
-                                constant:-20]
-      .active = YES;
-#endif
+
```
Comment (romoore) [inline components/TextFields/examples/TextFieldStatesViewController.m]: Nit: Remove extra newline

**Review by romoore — APPROVED** [review]
Looks good, with a few formatting nits.

File: components/Buttons/examples/supplemental/ButtonsTypicalUseSupplemental.m (hunk 1 above)
Comment (jverkoey) [reply components/Buttons/examples/supplemental/ButtonsTypicalUseSupplemental.m]: Done.

File: components/FlexibleHeader/src/MDCFlexibleHeaderView.m (hunk 2 above)
Comment (jverkoey) [reply components/FlexibleHeader/src/MDCFlexibleHeaderView.m]: Done.

File: components/FlexibleHeader/src/MDCFlexibleHeaderView.m (hunk 3 above)
Comment (jverkoey) [reply components/FlexibleHeader/src/MDCFlexibleHeaderView.m]: Done.

File: components/TextFields/examples/MultilineTextFieldLegacyExample.m (hunk 4 above)
Comment (jverkoey) [reply components/TextFields/examples/MultilineTextFieldLegacyExample.m]: Done.

File: components/TextFields/examples/TextFieldCustomFontExample.m (hunk 5 above)
Comment (jverkoey) [reply components/TextFields/examples/TextFieldCustomFontExample.m]: Done.

File: components/TextFields/examples/TextFieldOutlinedExample.m (hunk 6 above)
Comment (jverkoey) [reply components/TextFields/examples/TextFieldOutlinedExample.m]: Done.

File: components/TextFields/examples/TextFieldStatesViewController.m (hunk 7 above)
Comment (jverkoey) [reply components/TextFields/examples/TextFieldStatesViewController.m]: Done.

**Pull request merged**